erpnext/hr/report/salary_register/salary_register.py

```diff
@@ -117,6 +118,10 @@ def get_columns(salary_slips, filters):
 			"label": _("Salary Slip"), "fieldtype": "Link", "fieldname": "salary_slip_id","options": "Salary Slip",
 			"width": 80
 		},
+		{
+			"label": _("Payroll Entry"), "fieldtype": "Link", "fieldname": "payroll_entry", "options": "Payroll Entry",
+			"width": 80 if filters.show_employee_name else 140
+		},
```

Comment:
do not filter by payroll entry .. map filters from payroll entry to salary register instead